strapi/strapi
# POC: Add useEnterprise() hook; refactor settings router

### What does it do?

Introduces a new hook called `useEnterprise()`, which will be used to load EE data structures and maybe components at runtime.

#### TODO

- [x] Refactor settings router
- [x] Refactor app router
- [x] Refactor useSettingsMenu() hook
- [x] Audit logs link is now at the end of the list instead of the start
- [x] Add contributor docs

> **Note**
> It would be great to get some feedback on what you think about the approach. If you are all happy I'll start and write tests for `useEnterprise` and `useSettingsMenu` and document both hooks.

### Why is it needed?

We've decided we want to try this new approach because:

- it will allow runtime switching between CE and EE for the admin app
- it allows us to remove the `ee_else_ce` babel plugin, which is currently broken and not maintained
- it will allow us to get rid of the two-way coupling binding between CE and EE

### How to test?

For an end-user nothing should change - the settings menu should behave exactly the same as well as the SSO setup.
